packages/atomic/src/components/common/layout/layout-sections.spec.ts

```diff
@@ -0,0 +1,286 @@
+        (tagName) => {
+          const element = document.createElement(tagName);
+          Object.defineProperty(element, 'tagName', {
+            value: tagName.toUpperCase(),
+            writable: false,
+          });
+          expect(isResultSectionNode(element)).toBe(true);
+        }
+      );
+    });
+
+    describe('with product section elements', () => {
+      const productSectionTags: ItemSectionTagName[] = [
+        'atomic-product-section-visual',
+        'atomic-product-section-badges',
+        'atomic-product-section-actions',
+        'atomic-product-section-name',
+        'atomic-product-section-metadata',
+        'atomic-product-section-emphasized',
+        'atomic-product-section-description',
+        'atomic-product-section-bottom-metadata',
+        'atomic-product-section-children',
+      ];
+
+      it.each(productSectionTags)(
+        'should return true for %s element',
+        (tagName) => {
+          const element = document.createElement(tagName);
+          expect(isResultSectionNode(element)).toBe(true);
+        }
+      );
+    });
+
+    describe('with non-section elements', () => {
+      it('should return false for regular div element', () => {
+        const element = document.createElement('div');
+        expect(isResultSectionNode(element)).toBe(false);
+      });
+
+      it('should return false for span element', () => {
+        const element = document.createElement('span');
+        expect(isResultSectionNode(element)).toBe(false);
+      });
+
+      it('should return false for custom non-section element', () => {
+        const element = document.createElement('atomic-custom-element');
+        expect(isResultSectionNode(element)).toBe(false);
+      });
+
+      it('should return false for atomic result element that is not a section', () => {
+        const element = document.createElement('atomic-result-title');
+        expect(isResultSectionNode(element)).toBe(false);
+      });
+
+      it('should return false for atomic product element that is not a section', () => {
+        const element = document.createElement('atomic-product-title');
+        expect(isResultSectionNode(element)).toBe(false);
+      });
+    });
+
+    describe('with non-element nodes', () => {
+      it('should return false for text node', () => {
+        const textNode = document.createTextNode('Some text');
+        expect(isResultSectionNode(textNode)).toBe(false);
+      });
+
+      it('should return false for comment node', () => {
+        const commentNode = document.createComment('Some comment');
+        expect(isResultSectionNode(commentNode)).toBe(false);
+      });
+
+      it('should return false for document fragment', () => {
+        const fragment = document.createDocumentFragment();
+        expect(isResultSectionNode(fragment)).toBe(false);
+      });
+    });
+  });
+
+  describe('#containsSections', () => {
+    describe('with string content', () => {
```

Comment:
We could (should?) also be exhaustive here for the `true` cases (i.e., use a describe.each).

Otherwise the implementation of the function could theoretically change (e.g., additional values could return `true`, or some values that used to return `true` could start returning `false`) without the UTs breaking.

Using a describe.each just makes more sense overall I think for these tests.

We can just define the true / false tags we want to check at the top of the spec file, and reuse it as needed.